### PR TITLE
label unhealthy sessions

### PR DESCRIPTION
**Pull request by jackyzha0**
## Why

- we'd like an easy way to filter session close events by whether the session ended up being a healthy or unhealthy session to get some baseline SLAs

## What changed

- add `unhealthy-session` tags to all the places where it makes sense

## Versioning

- [ ] Breaking protocol change
- [ ] Breaking ts/js API change

### CODE REVIEW

File: transport/server.ts [hunk 1]
```diff
@@ -108,9 +108,12 @@ export abstract class ServerTransport<
     this.pendingSessions.delete(pendingSession);
   }
 
-  protected deleteSession(session: ServerSession<ConnType>): void {
+  protected deleteSession(
+    session: ServerSession<ConnType>,
+    unhealthy = false,
```
Comment (lhchavez) [inline transport/server.ts:113]: can we enum? `deletesession(foo, true)` doesn't tell me what `true` refers to. feels good but is bad.

File: transport/server.ts [hunk 2]
```diff
@@ -511,7 +514,7 @@ export abstract class ServerTransport<
               type: ProtocolError.InvalidMessage,
               message: reason,
             });
-            this.deleteSession(connectedSession);
+            this.deleteSession(connectedSession, true);
```
Comment (lhchavez) [inline transport/server.ts:517]: is this triggered in the out-of-order case?

File: transport/transport.ts [hunk 3]
```diff
@@ -201,8 +201,11 @@ export abstract class Transport<ConnType extends Connection> {
   }
 
   // state transitions
-  protected deleteSession(session: Session<ConnType>) {
-    session.log?.info(`closing session ${session.id}`, session.loggingMetadata);
+  protected deleteSession(session: Session<ConnType>, unhealthy = false) {
+    session.log?.info(`closing session ${session.id}`, {
+      ...session.loggingMetadata,
+      tags: unhealthy ? ['unhealthy-session'] : [],
```
Comment (lhchavez) [inline transport/transport.ts:207]: do we need to concat this with `session.loggingMetadata.tags ?? []`?

**Review by lhchavez — APPROVED** [review]

File: transport/transport.ts (hunk 3 above)
Comment (jackyzha0) [reply transport/transport.ts:207]: ah yes

File: transport/server.ts (hunk 2 above)
Comment (jackyzha0) [reply transport/server.ts:517]: not anymore, this is just failure to parse a message (we no longer delete a session on out-of-order as of https://github.com/replit/river/pull/244)

File: transport/server.ts (hunk 1 above)
Comment (masad-frost) [reply transport/server.ts:113]: Or "options" object

**Pull request merged**